src/controller/ai_controller.c

```diff
@@ -1533,14 +1655,355 @@ bool attempt_attack(controller *ctrl, bool highest_damage) {
+            // Speed Kick : D,F+K
+            controller_cmd(ctrl, ACT_DOWN, ev);
+            controller_cmd(ctrl, (o->direction == OBJECT_FACE_RIGHT ? ACT_DOWN|ACT_RIGHT : ACT_DOWN|ACT_LEFT), ev);
+            controller_cmd(ctrl, (o->direction == OBJECT_FACE_RIGHT ? ACT_RIGHT : ACT_LEFT), ev);
+            controller_cmd(ctrl, (o->direction == OBJECT_FACE_RIGHT ? ACT_RIGHT|ACT_KICK : ACT_LEFT|ACT_KICK), ev);
+            controller_cmd(ctrl, ACT_KICK, ev);
+            break;
+        case HAR_PYROS:
+            DEBUG("\e[35mPyros move:\e[0m Fire Spin");
+            // Fire Spin : D+P
+            controller_cmd(ctrl, ACT_DOWN, ev);
+            controller_cmd(ctrl, ACT_STOP, ev);
+            controller_cmd(ctrl, ACT_PUNCH, ev);
+            break;
+        case HAR_ELECTRA:
+            DEBUG("\e[35mElectra move:\e[0m Electric Shards");
+            // Electric Shards : D,F+P
+            controller_cmd(ctrl, ACT_DOWN, ev);
+            controller_cmd(ctrl, (o->direction == OBJECT_FACE_RIGHT ? ACT_DOWN|ACT_RIGHT : ACT_DOWN|ACT_LEFT), ev);
+            controller_cmd(ctrl, (o->direction == OBJECT_FACE_RIGHT ? ACT_RIGHT : ACT_LEFT), ev);
+            controller_cmd(ctrl, (o->direction == OBJECT_FACE_RIGHT ? ACT_RIGHT|ACT_PUNCH : ACT_LEFT|ACT_PUNCH), ev);
+            controller_cmd(ctrl, ACT_PUNCH, ev);
+            break;
+        case HAR_NOVA:
+            DEBUG("\e[35mNova move:\e[0m Earthquake Slam");
+            // Earthquake Slam : D,D,P
+            controller_cmd(ctrl, ACT_DOWN, ev);
+            controller_cmd(ctrl, ACT_STOP, ev);
+            controller_cmd(ctrl, ACT_DOWN, ev);
+            controller_cmd(ctrl, ACT_PUNCH, ev);
+            break;
+    }
+
+    return true;
+}
+
+/** 
+ * \brief Attempt to initiate a push atack using direct keyboard combinations.
+ *
+ * \param ctrl Controller instance.
+ * \param ev The current controller event.
+ *
+ * \return Boolean indicating whether an attack was initiated.
+ */
+bool attempt_trip_attack(controller *ctrl, ctrl_event **ev) {
+   // ai *a = ctrl->data;
+    object *o = ctrl->har;
+    har *h = object_get_userdata(o);
+
+    //int enemy_range = get_enemy_range(ctrl);
+
+    switch (h->state) {
+        case STATE_WALKTO:
+        case STATE_WALKFROM:
+        case STATE_CROUCHBLOCK:
+        case STATE_CROUCHING:
+            controller_cmd(ctrl, ACT_STOP, ev);
+            break;
+        case STATE_STANDING:
+            break;
+        default:
+            return false;
+    }
+
+    switch (h->id) {
+        case HAR_JAGUAR:
+        default:
+            DEBUG("\e[35mHar move:\e[0m Trip");
+            // Standard Trip : D+B+K
+            controller_cmd(ctrl, ACT_DOWN, ev);
+            controller_cmd(ctrl, (o->direction == OBJECT_FACE_RIGHT ? ACT_DOWN|ACT_LEFT : ACT_DOWN|ACT_RIGHT), ev);
+            controller_cmd(ctrl, (o->direction == OBJECT_FACE_RIGHT ? ACT_LEFT|ACT_KICK : ACT_RIGHT|ACT_KICK), ev);
+            controller_cmd(ctrl, ACT_KICK, ev);
+        break;
```

Comment:
inconsistent indentation, but also if there is only one case in a switch, this should not be a switch